bucket_map/src/index_entry.rs

```diff
@@ -84,3 +91,40 @@ impl IndexEntry {
         s.finish().max(1u64)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    impl IndexEntry {
+        pub fn new(key: Pubkey) -> Self {
+            IndexEntry {
+                key,
+                ref_count: 0,
+                storage_offset: 0,
+                storage_capacity_when_created_pow2: 0,
+                num_slots: 0,
+            }
+        }
+    }
+    #[test]
+    fn test_shared() {
```

Comment:
I don't understand the goal of this test. Can you help?

Reply:
added a comment. the next pr will be combining these 2 fields into 1. We need a test to ensure that the values are correct and independent of each other over their range.